xline/src/storage/kvwatcher.rs

```diff
@@ -174,24 +164,19 @@ impl WatcherMap {
     }
 }
 
-impl<S> KvWatcher<S>
-where
-    S: StorageApi,
-{
-    /// New `KvWatcher`
-    pub(super) fn new(
-        storage: Arc<KvStoreBackend<S>>,
-        mut kv_update_rx: mpsc::Receiver<(i64, Vec<Event>)>,
-    ) -> Self {
-        let inner = Arc::new(KvWatcherInner::new(storage));
-        let inner_clone = Arc::clone(&inner);
-        let _handle = tokio::spawn(async move {
-            while let Some(updates) = kv_update_rx.recv().await {
-                inner_clone.handle_kv_updates(updates).await;
-            }
-        });
-        Self { inner }
-    }
+/// Boot up a watcher task and return an `Arc<KvWatcher<S>>`
+pub(crate) fn watcher<S: StorageApi>(
+    storage: Arc<KvStore<S>>,
+    mut kv_update_rx: mpsc::Receiver<(i64, Vec<Event>)>,
+) -> Arc<KvWatcher<S>> {
+    let kv_watcher = Arc::new(KvWatcher::new(storage));
+    let watcher = Arc::clone(&kv_watcher);
+    let _handle = tokio::spawn(async move {
+        while let Some(updates) = kv_update_rx.recv().await {
+            watcher.handle_kv_updates(updates).await;
+        }
+    });
+    kv_watcher
 }
```

Comment:
Why not implement this method directly for `KvWatcher`

Reply:
Done